robottelo/hosts.py

```diff
@@ -1056,19 +1057,54 @@ def url_katello_ca_rpm(self):
         pub_url = urlunsplit(('http', self.hostname, 'pub/', '', ''))  # use url with https?
         return urljoin(pub_url, 'katello-ca-consumer-latest.noarch.rpm')
 
-    def capsule_certs_generate(self, capsule, **extra_kwargs):
+    def capsule_certs_generate(self, capsule, cert_path=None, **extra_kwargs):
         """Generate capsule certs, returning the cert path and the installer command args"""
         command = InstallerCommand(
             command='capsule-certs-generate',
             foreman_proxy_fqdn=capsule.hostname,
-            certs_tar=f'/root/{capsule.hostname}-certs.tar',
+            certs_tar=cert_path or f'/root/{capsule.hostname}-certs.tar',
             **extra_kwargs,
         )
         result = self.execute(command.get_command())
         install_cmd = InstallerCommand.from_cmd_str(cmd_str=result.stdout)
         install_cmd.opts['certs-tar-file'] = f'/root/{capsule.hostname}-certs.tar'
         return install_cmd
 
+    def custom_cert_generate(self, capsule_hostname):
+        """copy all configuration files to satellite host for generating custom certs"""
+        self.execute(f'mkdir ssl-build/{capsule_hostname}')
+        for file in [
+            'generate-ca.sh',
+            'generate-crt.sh',
+            'openssl.cnf',
+            'certs.sh',
+            'extensions.txt',
+        ]:
+            self.session.sftp_write(get_data_file(file), f'/root/{file}')
+        self.execute('echo 100001 > serial')
+        self.execute('bash generate-ca.sh')
+        self.execute(f'yes | bash generate-crt.sh {self.hostname}')
+        self.execute('bash certs.sh')
```

Comment:
Can you either check the exit code or some state on the system to validate whether this method did in fact do everything it was support to before returning?